File: tui/src/styles/theme.ts
// Obsidian-flavored dark palette. Lavender purple accent, deeper purple
// selection highlight, dark gray surfaces with a visibly lifted sidebar.
export const palette = {
  bg:            "#181818",
  bgLifted:      "#232323",
  boxBorder:     "#3a3a3a",
  accent:        "#a78bfa",
  accentDeep:    "#7c3aed",
  text:          "#dcddde",
  dim:           "#9e9e9e",
  dimmer:        "#5e5e5e",
  thinkingAmber: "#fab283",
  statusGreen:   "#3fb950",
  statusRed:     "#f85149",
  warning:       "#f5a742",
  mdCodeBg:      "#1f1f1f",
} as const;

// Semantic tokens — components MUST read from these, not from hex literals.
export const tokens = {
  // Surfaces
  bg: palette.bg,
  sidebarBg: palette.bgLifted,

  // Box / border / accent
  boxBorder: palette.boxBorder,
  accent: palette.accent,
  accentDeep: palette.accentDeep,

  // Text hierarchy
  textPrimary: palette.text,
  textDim: palette.dim,
  textDimmer: palette.dimmer,
  textInverse: palette.text,

  // Cursor / selection — strong purple for selection highlights
  cursorBg: palette.accentDeep,

  // Status / state
  toolRunning: palette.accent,
  toolDone:    palette.statusGreen,
  toolFailed:  palette.statusRed,
  toolBg:      palette.bg,
  toolHeader:  palette.accent,
  statusGreen: palette.statusGreen,
  statusRed:   palette.statusRed,

  // Accents
  thinkingAmber: palette.thinkingAmber,
  warning: palette.warning,

  // Markdown
  mdHeading:    palette.text,
  mdStrong:     palette.text,
  mdEm:         palette.text,
  mdCode:       palette.text,
  mdCodeBg:     palette.mdCodeBg,
  mdLink:       palette.accent,
  mdBlockquote: palette.dim,
  mdHr:         palette.boxBorder,
  mdListBullet: palette.accent,
} as const;

File: tui-go/internal/theme/theme.go
// Package theme is the Obsidian-flavored dark palette. Lavender purple accent,
// deeper purple selection highlight, dark gray surfaces with a visibly lifted
// sidebar. Mirrors tui/src/styles/theme.ts — components MUST read from these,
// not from hex literals.
package theme

import "github.com/charmbracelet/lipgloss"

var (
	Bg            = lipgloss.Color("#181818")
	BgLifted      = lipgloss.Color("#232323")
	BoxBorder     = lipgloss.Color("#3a3a3a")
	Accent        = lipgloss.Color("#a78bfa")
	AccentDeep    = lipgloss.Color("#7c3aed")
	Text          = lipgloss.Color("#dcddde")
	Dim           = lipgloss.Color("#9e9e9e")
	Dimmer        = lipgloss.Color("#5e5e5e")
	ThinkingAmber = lipgloss.Color("#fab283")
	StatusGreen   = lipgloss.Color("#3fb950")
	StatusRed     = lipgloss.Color("#f85149")
	Warning       = lipgloss.Color("#f5a742")
	MdCodeBg      = lipgloss.Color("#1f1f1f")

	// ConfigWizard-only colors.
	WizardLabel = lipgloss.Color("#9895b5")
	WizardValue = lipgloss.Color("#c9c5d9")
)

// PhaseColor maps a machine phase to its status-dot color. In-flight lifecycle
// phases pulse accent so the panel reads as "something is happening on this
// machine right now".
func PhaseColor(phase string) lipgloss.Color {
	switch phase {
	case "running":
		return StatusGreen
	case "sleeping":
		return Warning
	case "error":
		return StatusRed
	case "creating", "waking", "setup", "refreshing", "destroying":
		return Accent
	default: // none / destroyed / registered / unknown
		return Dimmer
	}
}
